clang/include/clang/Basic/Attr.td

```diff
@@ -1247,18 +1247,18 @@ def CUDAShared : InheritableAttr {
 }
 def : MutualExclusions<[CUDAConstant, CUDAShared, HIPManaged]>;
 
+def GlobalStorageNonLocalVar : SubsetSubject<Var,
+                                             [{S->hasGlobalStorage() &&
+                                               !S->isLocalVarDeclOrParm()}],
+                                             "global variables">;
+
 def SYCLDevice : InheritableAttr {
   let Spellings = [GNU<"sycl_device">];
-  let Subjects = SubjectList<[Function]>;
+  let Subjects = SubjectList<[Function, GlobalStorageNonLocalVar]>;
```

Comment:
Sorry I didn't notice this earlier but does a diagnostic trigger while parsing the attribute when it doesn't satisfy the conditions for  `GlobalStorageNonLocalVar`

Reply:
Yes, please see test [SemaSYCL/device_global_external.cpp](https://github.com/intel/llvm/pull/5963/files#diff-a79edb2ad29246c9fc79e3f70de77dfe2ad5539333ae87a5222334c489e5e843) on line 36. On attempt to apply the attribute to a local variable the diagnostic triggers.